inst/examples/variable.cpp

```diff
@@ -43,19 +43,44 @@ class variable {
 namespace dust {
 template <>
 struct has_gpu_support<variable> : std::true_type {};
+
+template <>
+size_t device_shared_size_int<variable>(dust::shared_ptr<variable> shared) {
+  return 1;
+}
+
+template <>
+size_t device_shared_size_real<variable>(dust::shared_ptr<variable> shared) {
+  return 2;
+}
+
+template <>
+void device_shared_copy<variable>(dust::shared_ptr<variable> shared,
+                                  int * shared_int,
+                                  variable::real_t * shared_real) {
+  typedef variable::real_t real_t;
+  shared_int = dust::shared_copy<int>(shared_int, shared->len);
+  shared_real = dust::shared_copy<real_t>(shared_real, shared->mean);
+  shared_real = dust::shared_copy<real_t>(shared_real, shared->sd);
```

Comment:
What would be the method to write this as a single H->D copy if we needed to do that? Could we 'stage' a std::vector on the host here, copy into it in parts like this, then copy the whole thing over to the device at the end?

Reply:
yes, exactly. That can be done in parallel too if we need to